### PR TITLE
Added helper methods in validator for the functions in Validation class

### LINKED ISSUE #7684: [RFC] Have real validation methods in Validator
I have seen many people struggle with how validators work in cake, and I think it is due to the fact that validation methods are actually in the `Validation` class instead of the `Validator`. I think we could have real methods in the validator for directly bridging to their implementation. For example:

```php
$validator
    ->naturalNumber('position')
    ->url('href', 'This is not a valid URL')
    ->between('salary', 1000, 2000);
```

The implementation of such rules can be done this way:

```php
public function url($field, $message, $when)
{
    return $this->add($field, ['rule' => 'url', 'message' => $message, 'on' => $when]);
}
```
I think this will make the validation rules more readable, easier to find and easier to document.

### PR DESCRIPTION
**Pull request by lorenzo**
This makes the validation API more discoverable and faster to create
validators. Fixes #7684

### CODE REVIEW

**Comment by lorenzo** [comment]
I will need a hand for adding the doc blocks to the new methods, who volunteers? :dog:

File: src/Validation/Validator.php [hunk 1]
```diff
@@ -535,6 +536,324 @@ public function notEmpty($field, $message = null, $when = false)
...
+        return $this->add($field, 'lengthBetween', $extra + [
+            'rule' => ['lengthBetween', array_shift($range), array_shift($range)],
+        ]);
+    }
+
+    public function creditCard($field, $type = 'all', $message = null, $when = null)
+    {
+        $extra = array_filter(['on' => $when, 'message' => $message]);
+        return $this->add($field, 'creditCard', $extra + [
+            'rule' => ['cc', $type, true],
```
Comment (markstory) [inline src/Validation/Validator.php]: The deep parameter (3rd param) is not even used in `Validation::luhn()`. I will clean that up separately.

File: src/Validation/Validator.php [hunk 2]
```diff
@@ -535,6 +536,324 @@ public function notEmpty($field, $message = null, $when = false)
...
+        return $this->add($field, 'utf8', $extra + [
+            'rule' => ['utf8', ['extended' => false]]
+        ]);
+    }
+
+    public function utf8Strict($field, $message = null, $when = null)
+    {
+        $extra = array_filter(['on' => $when, 'message' => $message]);
+        return $this->add($field, 'utf8Strict', $extra + [
+            'rule' => ['utf8', ['extended' => true]]
```
Comment (markstory) [inline src/Validation/Validator.php]: Extended isn't the same as 'strict'. Perhaps this should be `utf8Extended` or `utf8Emoji`?

Comment (lorenzo) [inline src/Validation/Validator.php]: Ah! I struggled finding a good name for this, I like your suggestion better

**Comment by markstory** [comment]
I can take care of adding API docs.

**Comment by lorenzo** [comment]
Thanks for the help @markstory

**Pull request merged**